index.js

```diff
@@ -44,11 +44,13 @@ function S3Readable(opts) {
     self.working -= 1
     debug("%s working %d queue %d %s", self.params.Key, self.working, self.queue._queue.length, result.range);
     self.done = result.data === null;
+    if (!self.done) self.done = result.contentLength < self.chunkSize;
     self.push(result.data);
   }
 }
 
 S3Readable.prototype._read = function(numBytes) {
+  if (this.done) return;
```

Comment:
This may be wrong, perhaps we should be pushing a zero-sized numBytes in the queue so that `sip` can callback `{data: null}`?

There are [some reports](https://github.com/jb55/s3-blob-store/issues/17) of the stream hanging and this is the only change that I can tell may be causing it? It's been a long time since I touched this code...

Reply:
Ok, but why revert this too? https://github.com/jb55/s3-blob-store/commit/c8cbe372f38b5e24bb5c1ce73293ffd6305a511b

Reply:
This is really important for us, we have a lot of small files :-(